File: src/etch/bindings/csharp/support/AsyncMode.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Support
{
    public enum AsyncMode
    {
        /// <summary>
        /// synchronous mode
        /// </summary>
        NONE,
        /// <summary>
        ///  the operation is queued to a thread pool for execution 
        /// </summary>
        QUEUED,
        /// <summary>
        ///  the operation is executed by a newly allocated thread 
        /// </summary>
        FREE       
    }
}

File: src/etch/bindings/java/support/AsyncMode.java
package etch.bindings.java.support;

/**
 * The asynchronous receiver mode of this helper.
 */
public enum AsyncMode
{
	/** synchronous mode */
	NONE,
	/** the operation is queued to a thread pool for execution */
	QUEUED,
	/** the operation is executed by a newly allocated thread */
	FREE
}